src/lib.rs

```diff
@@ -1106,17 +1117,17 @@ pub mod pallet {
 			Self::deposit_event(Event::NewReport {
 				query_id,
 				time: timestamp,
-				value,
+				value: value.clone(),
```

Comment:
As above, capture lengths into variables and pass to weights rather than cloning potentially heavy vecs.